.github/workflows/github_ci.yml

```diff
@@ -1,70 +1,115 @@
 name: Java CI
 
-on: [ pull_request, workflow_dispatch ]
+on:
+  pull_request:
+  workflow_dispatch:
 
 jobs:
+  build:
+    name: Build
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+      - name: Build with Gradle
+        run: ./gradlew assemble
 
-    build:
-        name: Build
-        strategy:
-            matrix:
-                os: [ ubuntu-latest, windows-latest, macos-latest ]
-        runs-on: ${{ matrix.os }}
-        steps:
-            -   name: Checkout
-                uses: actions/checkout@v3
-            -   name: Set up JDK 17
-                uses: actions/setup-java@v3
-                with:
-                    java-version: '17'
-                    distribution: 'temurin'
-                    cache: gradle
-            -   name: Build with Gradle
-                run: ./gradlew assemble
-    junit:
-        needs: build
-        name: JUnit
-        runs-on: ubuntu-latest
-        steps:
-            -   name: Checkout
-                uses: actions/checkout@v3
-            -   name: Set up JDK 17
-                uses: actions/setup-java@v3
-                with:
-                    java-version: '17'
-                    distribution: 'temurin'
-                    cache: gradle
-            -   name: JUnit
-                run: ./gradlew test
+  check:
+    name: Check files
+    outputs:
+        r: ${{steps.check_files.outputs.r}}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Check modified files
+        id: check_files
+        run: |
+          git diff --name-only HEAD^ HEAD > files.txt
```

Comment:
Was ist genau der Grund, nur die letzten beiden Commits zu betrachten? 

Könnte ich nicht im allerersten Commit eines Branches/PR die Änderungen an den Tests gemacht haben? Das würde ich dann nur 1x prüfen, und wenn dann noch weiter andere Commits kommen, würde das nie wieder wen interessieren ...

Reply:
Jetzt hast du genau den Gedanken den ich auch hatte. hihihih

Reply:
Ich glaube, ich würde wirklich `master..HEAD` diffen ... und wenn dabei was hochkommt, dann peng! Matrix.

Reply:
`git diff origin/master HEAD` sollte funktionieren, oder?

Reply:
ja, sollte gehen. kannst ja mal lokal testen "git diff origin/master..HEAD". und selbst wenn die action den master nicht auscheckt, sollte es den master als origin/master geben.

mir ist aber auch dunkel in erinnerung, dass github genau dafür bereits eine funktion oder was hat. hmmm.

Reply:
es gibt `git diff master` aber das ist hat der lokale master